includes/payment-methods/class-wc-stripe-express-checkout-element.php

```diff
@@ -430,4 +431,17 @@ public function display_express_checkout_button_separator_html() {
 		<p id="wc-stripe-express-checkout-button-separator" style="margin-top:1.5em;text-align:center;display:none;">&mdash; <?php esc_html_e( 'OR', 'woocommerce-gateway-stripe' ); ?> &mdash;</p>
 		<?php
 	}
+
+	/**
+	 * Determine whether to filter the cart needs shipping address.
+	 *
+	 * @param boolean $needs_shipping_address Whether the cart needs a shipping address.
+	 */
+	public function filter_cart_needs_shipping_address( $needs_shipping_address ) {
+		if ( $this->express_checkout_helper->has_subscription_product() && wc_get_shipping_method_count( true, true ) === 0 ) {
+			return false;
+		}
+
+		return $needs_shipping_address;
+	}
```

Comment:
@Mayisha What behavior do we want from ECE when there are no defined shipping zones, but the cart is shippable? Should we let the order push through with just the billing address? Do you know if there any implications when a shippable order does not have a shipping address?

Asking also because I think I introduced a bug in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3560 for block cart and block checkout, when using ECE and there are no defined shipping zones. It should be an easy fix but I wanted to make sure it's the behavior we want.

Reply:
My understanding is, if we are offering Stripe card payment method on the cart/checkout page and the displayed/charged amount on the Google/Apple Pay modal is correct, then we should offer Google/Apple Pay as well. 
What are your thoughts on this @annemirasol ? 👀

Reply:
Yeah, this makes sense. If we are letting it happen via other payment methods, we should let it happen via express checkout 👍